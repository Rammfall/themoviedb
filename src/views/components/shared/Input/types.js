import PropTypes from 'prop-types'

export const propTypes = {
  placeholder: PropTypes.string,
  validateStatus: PropTypes.oneOf([
    'success',
    'warning',
    'error',
    'validating',
    ''
  ]),
  validationMessage: PropTypes.string,
  type: PropTypes.oneOf(['password', 'text']),
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onBlur: PropTypes.func.isRequired,
  name: PropTypes.string,
  prefix: PropTypes.node
}

export const defaultProps = {
  placeholder: undefined,
  validateStatus: undefined,
  validationMessage: undefined,
  type: 'text',
  value: '',
  name: '',
  prefix: undefined
}
