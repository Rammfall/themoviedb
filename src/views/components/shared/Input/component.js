import React from 'react'
import { Form, Input as AntInput } from 'antd'

import { propTypes, defaultProps } from './types'

const Input = ({
  placeholder,
  validateStatus,
  validationMessage,
  type,
  value,
  onChange,
  onBlur,
  name,
  prefix,
  iconRender
}) => {
  const isPassword = type === 'password'
  const CustomInput = isPassword ? AntInput.Password : AntInput

  return (
    <Form.Item
      validateStatus={validateStatus}
      help={validationMessage}
    >
      <CustomInput
        placeholder={placeholder}
        type={type}
        onBlur={onBlur}
        prefix={prefix}
        onChange={onChange}
        name={name}
        value={value}
        iconRender={iconRender}
      />
    </Form.Item>
  )
}

Input.propTypes = {
  ...propTypes
}

Input.defaultProps = {
  ...defaultProps
}

export default Input
