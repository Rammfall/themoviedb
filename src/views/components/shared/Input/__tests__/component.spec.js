import React from 'react'
import { shallow } from 'enzyme'

import Input from '../component'

describe('Input component matches snapshot', () => {
  const defaultProps = {
    placeholder: 'test',
    onBlur: jest.fn(),
    onChange: jest.fn()
  }

  describe('if type text', () => {
    it('with default props', () => {
      // eslint-disable-next-line react/jsx-props-no-spreading
      const component = shallow(<Input {...defaultProps} />)

      expect(component).toMatchSnapshot()
    })
  })

  describe('if type password', () => {
    const passwordProps = {
      ...defaultProps,
      type: 'password'
    }

    it('with default props', () => {
      // eslint-disable-next-line react/jsx-props-no-spreading
      const component = shallow(<Input {...passwordProps} />)

      expect(component).toMatchSnapshot()
    })
  })
})
