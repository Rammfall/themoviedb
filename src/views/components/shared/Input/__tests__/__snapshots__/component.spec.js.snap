// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component matches snapshot if type password with default props 1`] = `
<FormItem>
  <Password
    action="click"
    iconRender={[Function]}
    name=""
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    placeholder="test"
    type="password"
    value=""
    visibilityToggle={true}
  />
</FormItem>
`;

exports[`Input component matches snapshot if type text with default props 1`] = `
<FormItem>
  <Input
    name=""
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    placeholder="test"
    type="text"
    value=""
  />
</FormItem>
`;
